torchsynth/module.py

```diff
@@ -618,31 +620,34 @@ class Noise(SynthModule):
     these in a synth then choose different seeds for each instance.
     """
 
-    def __init__(self, synthglobals: SynthGlobals, seed: int, **kwargs):
-        super().__init__(synthglobals, **kwargs)
+    noise_batch_size: Final[int] = 64
+
+    def __init__(self, synthconfig: SynthConfig, seed: int, **kwargs):
+        super().__init__(synthconfig, **kwargs)
 
         # Pre-compute default batch size number of noise samples
         generator = torch.Generator(device="cpu").manual_seed(seed)
-        noise = torch.empty((DEFAULT_BATCH_SIZE, self.buffer_size), device="cpu")
+        noise = torch.empty((self.noise_batch_size, self.buffer_size), device="cpu")
         noise.data.uniform_(-1.0, 1.0, generator=generator)
 
         # Repeat the noise batches if the current batch size is larger than the default
-        if self.batch_size > DEFAULT_BATCH_SIZE:
-            if self.batch_size % DEFAULT_BATCH_SIZE != 0:
+        if self.batch_size > self.noise_batch_size:
+            if self.batch_size % self.noise_batch_size!= 0:
                 raise ValueError(
                     "Batch size is not divisible by the default "
-                    f"batch size of {DEFAULT_BATCH_SIZE}"
+                    f"batch size of {self.noise_batch_size}"
                 )
 
-            noise = noise.repeat(self.batch_size // DEFAULT_BATCH_SIZE, 1)
+            noise = noise.repeat(self.batch_size // self.noise_batch_size, 1)
 
         self.register_buffer("noise", noise.to(self.device))
         self.offset = 0
 
     def _forward(self) -> Signal:
         # Return noise quickly if we are returning the whole batch
         # and don't need to offset at all.
-        if self.batch_size >= DEFAULT_BATCH_SIZE and self.offset == 0:
```

Comment:
@jorshi is this actually what we want?

Reply:
Yah I think so - in init batch_size number of random noise samples are created. If batch_size is greater than the noise_batch_size (64) then it repeats the noise samples. So if the offset is zero and the batch_size is greater than or equal to the noise batch size then we can just return the buffer of noise.

Reply:
I admit that it is confusing. Would be good to clean this up and simplify